### PR TITLE
Improve test suite: remove global tests, increase net safety

### PR DESCRIPTION
**Pull request by Slayer95**

### CODE REVIEW

File: sockets.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ if (cluster.isMaster) {
 	var workers = exports.workers = {};
 
 	var spawnWorker = exports.spawnWorker = function () {
-		var worker = cluster.fork({PSPORT: Config.port, PSBINDADDR: Config.bindaddress || ''});
+		var worker = cluster.fork({PSPORT: Config.port, PSBINDADDR: Config.bindaddress || '', PSNOSSL: Config.ssl ? 0 : 1});
```
Comment (Zarel) [inline sockets.js]: Do we need `PSNOSSL`? If you do `node app.js 8001` you probably don't want SSL either. My current approach handles both situations.

Comment (Zarel) [inline sockets.js]: Oh, nevermind. This is an okay approach.

File: test/main.js [hunk 2]
```diff
@@ -18,6 +34,30 @@ before('initialization', function () {
...
 	BattleEngine.Battle.prototype.receive = function () {};
+
+	callback();
+}
+
+before('initialization', function (done) {
+	this.timeout(0); // Remove timeout limitation
+
+	// Load and override configuration before starting the server
+	var Config = require('./../config/config.js');
```
Comment (Zarel) [inline test/main.js]: The Travis error is because of this line. Travis runs PS for the first time, before `config.js` exists.

Comment (codelegend) [inline test/main.js]: what if `config.js` does not exist? (the build error...)

Comment (Slayer95) [inline test/main.js]: Fixed

**Comment by Zarel** [comment]
So dev is crashing with:

```
CRASH: Error: bind EADDRINUSE
    at errnoException (net.js:904:11)
    at net.js:1084:30
    at Object.1:1 (cluster.js:594:5)
    at handleResponse (cluster.js:171:41)
    at respond (cluster.js:192:5)
    at handleMessage (cluster.js:202:5)
    at process.emit (events.js:117:20)
    at handleMessage (child_process.js:322:10)
    at child_process.js:396:7
    at process.handleConversion.net.Native.got (child_process.js:91:7)

  
CRASH: TypeError: Cannot read property 'port' of null
    at Server.<anonymous> (cluster.js:577:30)
    at Server.g (events.js:180:16)
    at Server.emit (events.js:92:17)
    at net.js:1055:10
    at process._tickCallback (node.js:419:13)


CRASH: AssertionError: false == true
    at ChildProcess.<anonymous> (child_process.js:372:7)
    at ChildProcess.emit (events.js:117:20)
    at handleMessage (child_process.js:322:10)
    at Pipe.channel.onread (child_process.js:349:11)
```

Do you know what's going on here?

**Comment by Zarel** [comment]
Also Travis CI is still pending for what seems to be all pullreqs since this one was accepted, which makes me nervous.

**Comment by Zarel** [comment]
https://github.com/Zarel/Pokemon-Showdown/compare/a379ff5fc485~8...a379ff5fc485

This is very worrisome.

File: sockets.js [hunk 3]
```diff
@@ -114,14 +114,8 @@ if (cluster.isMaster) {
 } else {
 	// is worker
 
-	var portOverridden = false;
-	if (process.env.PSPORT) {
-		if ((+Config.port || 8000) !== (+process.env.PSPORT)) {
-			portOverridden = true;
-		}
-		Config.port = +process.env.PSPORT;
```
Comment (Slayer95) [inline sockets.js]: Oops!

**Comment by Slayer95** [comment]
Fixed by https://github.com/Zarel/Pokemon-Showdown/commit/83ffa44e512af89f2fcbb44b1e5761f1905446a4
Regarding Travis CI pending builds, I no longer see that issue.

**Pull request merged**